woocommerce/woocommerce-ios
# [HACK week] Product form - Action buttons to mark/remove as favorite

Part of: #12274 


## Description

Show action buttons in product form to mark/remove a product as a favorite. We will show favorite products in the product list screen in a future PR.

Changes
- Feature flag to allow marking a product as favorite.
- Use case to store favorite product list into user defaults.
- Add action buttons to mark/remove a product as favorite.

## Testing instructions
- Turn on the `favoriteProducts` feature flag by returning `true` from [here](https://github.com/woocommerce/woocommerce-ios/pull/12284/files#diff-26c9afc49bb1f1a168517d43d5e36a64ff45adb046469369ef3e4d6800ccc19bR92)
- Login into the app
- Navigate to Products tab
- Open any product
- Tap "…" from the action sheet tap "Mark as favorite"
- From Xcode log ensure that `product_detail_mark_as_favorite_tapped` event is tracked
- Navigate back/Quit the app
- Open the same product again from the Products tab
- Tap "…" from the action sheet 
- You should see "Remove from favorite" option. Tap on it to remove the product from favorite list.
- From Xcode log ensure that `product_detail_remove_from_favorite_tapped` event is tracked
- Reopen the product and validate that you see "Mark as favorite"
- Switch stores and ensure that the product marked as favorite is still marked as favorite. 
- Log out and log into the same store again. Validate that the favorite products list is cleared after logout.

## Screenshots
![Simulator Screen Recording - iPhone 15 Pro - 2024-03-15 at 11 13 16](https://github.com/woocommerce/woocommerce-ios/assets/524475/f5a0e788-009a-497e-b252-b9c6900387e7)

---
